### ISSUE TITLE
Web3Auth.connect() should return immediately if LOGIN_MODAL_EVENTS.MODAL_VISIBILITY is emitted with 'false'.

### ISSUE THREAD

**Issue by lithdew** [Web3Auth/web3auth-web#1501]
### **Describe the bug**

Web3Auth.connect() should return immediately if LOGIN_MODAL_EVENTS.MODAL_VISIBILITY is emitted with 'false'.

### **To Reproduce**

Close the modal that pops up. connect() will not return.

### **Expected behavior**

connect() should throw an error.

### **Workaround**

```ts
let provider: Web3Auth["provider"] = null;

if (!web3auth.connected) {
    web3auth.loginModal.open();

    type Result =
        | { status: "connected"; provider: NonNullable<Web3Auth["provider"]> }
        | { status: "modal closed"; provider?: never; error?: never }
        | { status: "errored"; provider?: never; error: unknown };

    let emit: (result: Result) => void;

    let modalVisible = true;

    const modalVisibilityListener = (open: boolean) => {
        modalVisible = open;

        if (!open) {
            emit({ status: "modal closed" });
        }
    };

    const connectedListener = () => {
        if (!web3auth.provider) {
            throw new Error("web3auth.provider is null after being connected");
        }
        emit({ status: "connected", provider: web3auth.provider });
    };

    const errorListener = (err: unknown) => {
        emit({ status: "errored", error: err });
    };

    login: while (true) {
        web3auth.loginModal.on("MODAL_VISIBILITY", modalVisibilityListener);
        web3auth.on("connected", connectedListener);
        web3auth.on("errored", errorListener);

        const result = await new Promise<Result>((resolve) => (emit = resolve))
            .then((result) => {
                web3auth.loginModal.off("MODAL_VISIBILITY", modalVisibilityListener);
                web3auth.off("connected", connectedListener);
                web3auth.off("errored", errorListener);

                return result;
            })
            .catch((err: unknown) => {
                web3auth.loginModal.off("MODAL_VISIBILITY", modalVisibilityListener);
                web3auth.off("connected", connectedListener);
                web3auth.off("errored", errorListener);

                return { status: "errored" as const, error: err };
            });

        switch (result.status) {
            case "connected":
                provider = result.provider;
                break login;
            case "errored":
                console.warn(result.error);
                if (modalVisible) {
                    continue login;
                }
            case "modal closed":
                set(sessionAtom, { status: "init" });
                return;
        }
    }
}
```

**Comment by chaitanyapotti** [comment]
We're working on this. Thanks for the report

**Comment by r0kan** [comment]
Some issues

**Comment by Cherden** [comment]
Hi. Any update on when this problem will be tackled?

**Comment by Simse92** [comment]
bump

**Comment by chaitanyapotti** [comment]
Fixed in 7.3.0
Pls upgrade
